collector/library/pod.js

```diff
@@ -224,15 +210,57 @@ class Pod {
      * 
      */
     async submitDataToLogstash({ data, batch, }) {
-        const body = JSON.stringify({ message: data.toString() });
+        const dataAsString = data.toString();
+        let lines;
+
+        if (this.remainder.length > 0) {
```

Comment:
this if is redundant. you can always add remainder wether it's empty or not. just watch out for undefined...